bluesky/scientific_callbacks.py

```diff
@@ -0,0 +1,83 @@
+import numpy as np
+from scipy.ndimage import center_of_mass
+from bluesky.callbacks import CollectThenCompute
+
+
+class PeakStats(CollectThenCompute):
+
+    def __init__(self, x, y):
+        """
+        Compute peak statsitics after a run finishes.
+
+        Results are stored in the attributes.
+
+        Parameters
+        ----------
+        x : string
+            field name for the x variable (e.g., a motor)
+        y : string
+            field name for the y variable (e.g., a detector)
+
+        Attributes
+        ----------
+        com : center of mass
+        cen : TBD
+        max : x location of y maximum
+        min : x location of y minimum
+        """
+        self.x = x
+        self.y = y
+        self.com = None
+        self.cen = None
+        self.max = None
+        self.min = None
+        super().__init__()
+
+    def compute(self):
+        "This method is called at run-stop time by the superclass."
+        x = []
+        y = []
+        for event in self._events:
+            try:
```

Comment:
I would do this as

```python
try:
    _x = event['data'][self.x]
   _y = event['data'][self.y]
except KeyError:
    pass
else:
    x.append(_x)
    y.append(_y)
```

so that they never get out of sync